Update script to make it use OpsGenie's REST API

cglrkn opened the pull request:
The script is making API calls to the deprecated OpsGenie Web API, we updated the script to make it use new OpsGenie's REST API.

cglrkn committed:
Update script to make it use OpsGenie's REST API

The script is making API calls to the deprecated OpsGenie Web API, we updated the script to make it use new OpsGenie's REST API.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=9399) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=9399) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=9399) it.</sub>

bergquist commented on pkg/services/alerting/notifiers/opsgenie.go:
```diff
@@ -119,14 +121,17 @@ func (this *OpsGenieNotifier) closeAlert(evalContext *alerting.EvalContext) erro
 
 	cmd := &m.SendWebhookSync{
-		Url:        opsgenieCloseAlertURL,
+		Url:        fmt.Sprintf("%s/%s/close?identifierType=alias", opsgenieAlertURL,"alertId-"+strconv.FormatInt(evalContext.Rule.Id, 10)),
```
I think the string can be replaced with `fmt.Sprintf("%s/alertId-%d", opsgenieAlertURL, evalContext.Rule.Id)`

bergquist reviewed (changes requested):
Looks good. Just want some minor adjustments for building the url for closing the alert

cglrkn committed:
Build URL for close alert request differently

cglrkn commented:
I updated the code according to your request. However, we need to put /close?identifierType=alias to make a successful request. Thus, I build the first part as you suggested and added /close?identifierType=alias at the end.

bergquist commented:
@cglrkn Oh thanks for spotting that! :)

Thank you for contributing to Grafana

bergquist commented on pkg/services/alerting/notifiers/opsgenie.go:
```diff
@@ -119,14 +121,17 @@ func (this *OpsGenieNotifier) closeAlert(evalContext *alerting.EvalContext) erro
 
 	cmd := &m.SendWebhookSync{
-		Url:        opsgenieCloseAlertURL,
+		Url:        fmt.Sprintf("%s/alertId-%d/close?identifierType=alias", opsgenieAlertURL, evalContext.Rule.Id)
```
The line is missing a comma at the end. otherwise it wont compile

bergquist commented:
@cglrkn Oh thanks for spotting that! :)

Still a minor change needed for this PR to compile. Will merge once its fixed.

cglrkn committed:
Add the missing comma

cglrkn commented:
Added the comma, sorry for missing that.

bergquist commented:
No worries! 

The build is still failing. Seems like some formatting issue. Please run `gofmt -w ./pkg` in the grafana root folder.

cglrkn committed:
Fix formatting issue

cglrkn commented:
I run the command. Thanks! @bergquist

bergquist commented:
@cglrkn Awesome! thank you for contributing to Grafana! :)
